src/java.base/share/classes/java/lang/foreign/AddressLayout.java

```diff
@@ -0,0 +1,113 @@
+/*
+ *  Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ *  DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ *  This code is free software; you can redistribute it and/or modify it
+ *  under the terms of the GNU General Public License version 2 only, as
+ *  published by the Free Software Foundation.  Oracle designates this
+ *  particular file as subject to the "Classpath" exception as provided
+ *  by Oracle in the LICENSE file that accompanied this code.
+ *
+ *  This code is distributed in the hope that it will be useful, but WITHOUT
+ *  ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ *  FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ *  version 2 for more details (a copy is included in the LICENSE file that
+ *  accompanied this code).
+ *
+ *  You should have received a copy of the GNU General Public License version
+ *  2 along with this work; if not, write to the Free Software Foundation,
+ *  Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ *   Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ *  or visit www.oracle.com if you need additional information or have any
+ *  questions.
+ *
+ */
+
+package java.lang.foreign;
+
+import jdk.internal.foreign.layout.ValueLayouts;
+import jdk.internal.javac.PreviewFeature;
+import jdk.internal.reflect.CallerSensitive;
+
+import java.lang.foreign.Linker.Option;
+import java.lang.invoke.MethodHandle;
+import java.nio.ByteOrder;
+import java.util.Optional;
+
+/**
+ * A value layout used to model the address of some region of memory. The carrier associated with an address layout is
+ * {@code MemorySegment.class}. The size and alignment of an address layout are platform dependent
+ * (e.g. on a 64-bit platform, the size and alignment of an address layout are set to 64 bits).
+ * <p>
+ * An address layout may optionally feature a {@linkplain #targetLayout() target layout}. The target layout of an address
+ * layout is used to model the layout of the region of memory whose address is described by that address layout.
+ * For instance, if an address layout has target layout {@link ValueLayout#JAVA_INT}, the region of memory pointed to by the address
+ * described by the address layout is 4 bytes long. Specifying a target layout can be useful in the following situations:
```

Comment:
Some redundant text here it seems
```suggestion
 * For instance, if an address layout has target layout {@link ValueLayout#JAVA_INT}, the region of memory 
 * described by the address layout is 4 bytes long. Specifying a target layout can be useful in the following situations:
```

Reply:
Not sure about this - (I went a bit back and forth with the javadoc text). An address layout describe an address value, not a memory region. E.g. it describes 64 bits. With a target layout you can say that those 64 bits point to a memory region that has a given layout. It is a bit convoluted to describe.

Reply:
Gave it another stab - let me know what you think.